src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/CSharpCompilerOptionsTests.cs

```diff
@@ -102,32 +102,22 @@ public async Task ProjectOutputBinPathChange_CPS()
 
         [WpfFact, WorkItem(14520, "https://github.com/dotnet/roslyn/issues/14520")]
         [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
-        public async Task InvalidProjectOutputBinPaths_CPS1()
+        public async Task InvalidProjectOutputBinPaths_CPS()
         {
             using var environment = new TestEnvironment();
-            using var project1 = await CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test", binOutputPath: null);
-            // Null output path is allowed.
-            Assert.Null(project1.BinOutputPath);
-        }
-
-        [WpfFact, WorkItem(14520, "https://github.com/dotnet/roslyn/issues/14520")]
-        [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
-        public async Task InvalidProjectOutputBinPaths_CPS2()
-        {
-            using var environment = new TestEnvironment();
-            using var project2 = await CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test2", binOutputPath: String.Empty);
-            // Empty output path is not allowed, it gets reset to null.
-            Assert.Null(project2.BinOutputPath);
+            await Assert.ThrowsAsync<InvalidOperationException>(() => CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test2", binOutputPath: null));
+            await Assert.ThrowsAsync<InvalidOperationException>(() => CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test2", binOutputPath: String.Empty));
         }
 
         [WpfFact, WorkItem(14520, "https://github.com/dotnet/roslyn/issues/14520")]
         [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
         public async Task InvalidProjectOutputBinPaths_CPS3()
         {
             using var environment = new TestEnvironment();
-            using var project3 = await CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test3", binOutputPath: "Test.dll");
-            // Non-rooted output path is not allowed, it gets reset to a temp rooted path.
-            Assert.Equal(Path.Combine(Path.GetTempPath(), "Test.dll"), project3.BinOutputPath);
+
+            // Non-rooted output path is not allowed, the project system will handle the exception:
```

Comment:
So what's the plan here now for https://github.com/dotnet/roslyn/issues/14520? Does the project system handle that in some different way now? If nothing else, consider removing the WorkItem attribute since this is no longer covering that bug anymore.

Reply:
@drewnoakes Do you need a custom exception type here from us if you need to handle that differently?

Reply:
In this case an exception would be our bug, as we should only pass absolute paths. I think a custom exception would only be useful if we couldn't expect this exception and wanted to handle that case especially.

Reply:
I'm gonna add the custom exception as I think it would still be useful to distinguish between Roslyn bug (that results in generic InvalidOperationException) and invalid property value.